tests/run/generators_py.py

```diff
@@ -387,3 +387,20 @@ def test_yield_in_const_conditional_true():
     """
     if True:
         print((yield 1))
+
+
+def test_generator_scope():
+    """
+    Tests that the function is run at the correct time
+    (i.e. when the generator is created, not when it's run)
+    >>> list(test_generator_scope())
+    inner running
+    generator created
+    [0, 10]
+    """
+    def inner(val):
+        print("inner running")
+        return [0, val]
+    gen = (a for a in inner(10))
+    print("generator created")
+    return gen
```

Comment:
I think this test would be easier to read if `10` were an argument of the test itself.
```suggestion
def test_generator_scope(n):
    """
    Tests that the function is run at the correct time
    (i.e. when the generator is created, not when it's run)
    >>> list(test_generator_scope(10))
    inner running
    generator created
    [0, 10]
    """
    def inner(val):
        print("inner running")
        return [0, val]
    gen = (a for a in inner(n))
    print("generator created")
    return gen
```

Reply:
Interestingly, the code generated will somewhat different - in my version `10` is inlined directly into the C code while in your version it's captured. It shouldn't affect the test of course.

I'm not sure I see a difference in readability so I'll ignore the suggestion though

Reply:
It's harder to see where the `10` is coming from when just reading the test from top to bottom and having to basically unravel the test completely to figure that out. Not a huge deal, but still a perceptible improvement, IMO.